index.js

```diff
@@ -48,60 +48,218 @@ const player = {
     { id: 5, name: 'Israeli', songs: [4, 5] },
   ],
   playSong(song) {
-    console.log(/* your code here */)
-  },
+    console.log(`Playing ${song.title} from ${song.album} by ${(song.artist)} | ${mmssDuration(song.duration)}.`);
+  }
 }
+const mmssDuration=(duration)=> {
+    let minutes = Math.floor(duration / 60);
+    let seconds = duration % 60;
+    if (minutes < 10) {
+      minutes = "0" + minutes;
+    }
+    if (seconds < 10) {
+      seconds = "0" + seconds;
+    }
+    return minutes.toString() + ':' + seconds.toString();
+  }
 
-function playSong(id) {
-  // your code here
-}
-
-function removeSong(id) {
-  // your code here
-}
-
-function addSong(title, album, artist, duration, id) {
-  // your code here
-}
-
-function removePlaylist(id) {
-  // your code here
-}
-
-function createPlaylist(name, id) {
-  // your code here
-}
-
-function playPlaylist(id) {
-  // your code here
-}
+  const getSongById= (id)=>{
+    for (let num in player.songs) { 
+      if (player.songs[num].id === id){ 
+        return player.songs[num]; 
+    }
+  }
+    throw new Error("No song was found");
+  }
 
-function editPlaylist(playlistId, songId) {
-  // your code here
-}
+  function playSong(id) {
+      if(getSongById(id)===undefined){
+      throw " ${id} ID not exists";
+    }
+    return player.playSong(getSongById(id));
+  }
+  
+  function removeSong(id) {
+    if(getSongById(id)===undefined){
```

Comment:
you are using the getSongById(id) value 2 time in this function so it should be a variable